MCprep_addon/materials/generate.py

```diff
@@ -1112,6 +1125,55 @@ def texgen_seus(mat: Material, passes: Dict[str, Image], nodeInputs: List, use_r
 	# nodeTexDisp["MCPREP_disp"] = True
 	nodeTexDiff.image = image_diff
 
+def generate_base_material(context: Context, name: str, path: Union[Path, str], useExtraMaps):
+		"""Generate a base material from name and active resource pack"""
+		try:
+			image = bpy.data.images.load(path, check_existing=True)
+		except: # if Image is not found
+			image = None
+		mat = bpy.data.materials.new(name=name)
+
+		engine = context.scene.render.engine
+		if engine == 'CYCLES' or engine == 'BLENDER_EEVEE':
+			# need to create at least one texture node first, then the rest works
+			mat.use_nodes = True
+			nodes = mat.node_tree.nodes
+			node_diff = create_node(
+				nodes, 'ShaderNodeTexImage', 
+				name="Diffuse Texture",
+				label="Diffuse Texture", 
+				location=(-380, 140),
+				interpolation='Closest', 
+				image=image
+			)
+			node_diff["MCPREP_diffuse"] = True
+
+			# The offset and link diffuse is for default no texture setup
+			links = mat.node_tree.links 
+			principled = None
+			for n in nodes:
+				if n.bl_idname == 'ShaderNodeBsdfPrincipled':
+					principled = n 
+					break
+			
+			links.new(node_diff.outputs[0], principled.inputs[0])
+			links.new(node_diff.outputs[1], principled.inputs["Alpha"])
```

Comment:
The variable here is redundant in this case, linking can be done in the actual for loop itself, and the variable is only used for links anyway.

 ```suggestion
			for n in nodes:
				if n.bl_idname == 'ShaderNodeBsdfPrincipled':			
			        links.new(node_diff.outputs[0], principled.inputs[0])
			        links.new(node_diff.outputs[1], principled.inputs["Alpha"])
			        break
```